### PR TITLE
Add DEVELOPMENT guide

### PR DESCRIPTION
**Pull request by chaodaiG**
Something we've been missing for a long time

/assign @chizhg @albertomilan

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Development
+
+This doc explains how to setup a development environment so you can get started
+[contributing](https://www.knative.dev/contributing/) to `Knative Test Infra`. Also
+take a look at:
```
Comment (mattmoor) [inline DEVELOPMENT.md:5]: Format markdown:
```suggestion
[contributing](https://www.knative.dev/contributing/) to `Knative Test Infra`.
Also take a look at:
```

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+As you make changes to the code-base, there are two special cases to be aware
+of:
+
+- **If you change an input to generated code**, then you must run
+  [`./hack/update-codegen.sh`](./hack/update-codegen.sh). Inputs include:
+
+  - Prow configs templates
+    [config/prod/prow/config_knative.yaml](./config/prod/prow/config_knative.yaml).
+
+  - Prow configs generater [tools/config-generator](./tools/config-generator).
```
Comment (albertomilan) [inline DEVELOPMENT.md:54]: Typo on `generater`

Comment (chaodaiG) [reply DEVELOPMENT.md]: Fixed, thanks!

File: DEVELOPMENT.md (hunk 1 above)
Comment (mattmoor) [inline DEVELOPMENT.md:5]: Format markdown:
```suggestion
[contributing](https://www.knative.dev/contributing/) to `Knative Test Infra`.
Also take a look at:
```

Comment (albertomilan) [reply DEVELOPMENT.md:5]: @chaodaiG, please confirm if you plan to follow this suggestion. Either way, the change looks good to me

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Before submitting a PR, see also [CONTRIBUTING.md](./CONTRIBUTING.md).
+
+### Sign up for GitHub
+
+Start by creating [a GitHub account](https://github.com/join), then setup
+[GitHub access via SSH](https://help.github.com/articles/connecting-to-github-with-ssh/).
+
+### Checkout your fork
+
+The Go tools require that you clone the repository to the
```
Comment (chizhg) [inline DEVELOPMENT.md:21]: This is actually not a requirement anymore after we switched to Go modules.

Comment (chaodaiG) [reply DEVELOPMENT.md:21]: Does this mean the `path_alias` in prow jobs isn't required either?

Comment (chizhg) [reply DEVELOPMENT.md:21]: I _think_ it's not required either, but I'm not entirely sure. We may need to do some quick experiment to see if that's the case.

Comment (chizhg) [reply DEVELOPMENT.md:21]: I've made a Prow job without path_alias - https://prow.knative.dev/view/gcs/knative-prow/logs/ci-knative-serving-continuous-test-noalias/1280208283721797632, let's see how it goes.

Comment (chaodaiG) [reply DEVELOPMENT.md:21]: Looks like this means someone will need to download knative.dev/test-infra for building the code, which is odd to me:
```
go: downloading knative.dev/test-infra v0.0.0-20200630141629-15f40fe97047
```

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Development
+
+This doc explains how to setup a development environment so you can get started
+[contributing](https://www.knative.dev/contributing/) to `Knative Test Infra`. Also
+take a look at:
```
Comment (mattmoor) [inline DEVELOPMENT.md:5]: Format markdown:
```suggestion
[contributing](https://www.knative.dev/contributing/) to `Knative Test Infra`.
Also take a look at:
```

File: DEVELOPMENT.md (hunk 3 above)
Comment (chaodaiG) [reply DEVELOPMENT.md]: nvm, didn't notice the example was serving repo. Updated the instruction

**Pull request merged**